plugins/modules/job_invocation.py

```diff
@@ -0,0 +1,223 @@
+---
+module: job_invocation
+short_description: Manage Job Invocations in Foreman
+description:
+  - "Manage Remote Execution Job Invocations"
+author:
+  - "Peter Ondrejka (@pondrejk)"
+options:
+  search_query:
+    description:
+      - Search query to identify hosts
+    type: str
+  bookmark_id:
+    description:
+      - Bookmark ID to infer the search query from
+    choices:
+      - 6 (active)
+      - 9 (disabled)
+      - 8 (error)
+      - 1 (list hypervisors)
+      - 10 (ok hosts)
+      - 7 (out of sync)
+    type: int
+  job_template_id:
+    description:
+      - ID of the job template
+    choices:
+      - 148 (Run Command - SSH default)
+      - 177 (Run Command - Ansible default)
+    required: true
+    type: int
+  targeting_type:
+    description:
+      - Dynamic query updates the search results before execution (useful for scheduled jobs)
+    choices:
+      - static_query (default)
+      - dynamic_query
+    type: str
+  randomized_ordering:
+    description:
+      - Whether to order the selected hosts randomly, default false (alphabetical)
+    type: bool
+  execution_timeout_interval:
+    description:
+      - Override the timeout interval from the template for this invocation only
+    type: int
+  ssh:
+    description:
+      - ssh related options
+    type: dict
+    suboptions:
+      effective_user:
+      description:
+        - What user should be used to run the script (using sudo-like mechanisms)
+        - Defaults to a template parameter or global setting
+      type: str
+  command:
+    description:
+      - Command to be executed on host. Required for command templates
+    type: str
+  recurrence:
+    description:
+      - Schedule a recurring job
+    type: dict
+    suboptions:
+      cron_line:
+        description:
+          - How often the job should occur, in the cron format
+        type: str
+      max_iteration:
+        description:
+          - Repeat a maximum of N times
+        type: int
+      end_time:
+        description:
+          - Perform no more executions after this time
+        type: datetime
+  scheduling:
+    description:
+      - Schedule the job to start at a later time
+    type: dict
+    suboptions:
+      start_at:
+        description:
+          - Schedule the job for a future time
+        type: datetime
+      start_before:
+        description:
+          - Indicates that the action should be cancelled if it cannot be started before this time.
+        type: datetime
+  concurrency_control:
+    description:
+      - Control concurrency level and distribution over time
+    type: dict
+    suboptions:
+      time_span:
+        description:
+          - Distribute tasks over given number of seconds
+        type: int
+      concurrency_level:
+        description:
+          - Maximum jobs to be executed at once
+        type: int
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.entity_state_with_defaults
```

Comment:
```suggestion
  - theforeman.foreman.foreman.entity_state
```
You don't have `present_with_defaults`.